pkg/components/header.go

```diff
@@ -8,38 +8,38 @@ import (
 	"github.com/asecurityteam/transport"
 )
 
-// HeaderConfig configures automated header injection.
-type HeaderConfig struct {
```

Comment:
task: We use this plugin so this will break something. The library is still pre-1.0 so I'm OK with making breaking changes but we'll need to search around to make sure we don't break a deployment.

Reply:
Will check.  Hopefully whomever is using the headerinject plugin is doing so by referring to a specific version of transportd rather than `:latest`.

Reply:
Thanks @kconwayatlassian for having a look around.  I did the same.  Our conclusion is that `headerinject` _was_ formerly used, but has since been replaced by `basicauth` in all the places it was used.  So, we assume very little to no risk.